### PR TITLE
feat(protractor): add e2e tests in CI

### LINKED ISSUE #732: Protractor CI
Protractor + Travis CI + <s>Sauce Labs</s>

### PR DESCRIPTION
**Pull request by Shyam-Chen**
close #732

### CODE REVIEW

**Comment by Shyam-Chen** [comment]
Appveyor
```bash
[02:04:45] Starting 'e2e'...
Using ChromeDriver directly...
[launcher] Running 1 instances of WebDriver
Spec started
Started
 
  App
    ✓ should have a title
.    ✓ should have <nav>
.    ✓ should have correct nav text for Home
.    ✓ should have correct nav text for About
. 
  About
    ✗ should have correct feature heading
      - Failed: No element found using locator: By(css selector, sd-app sd-about h2)
F 
  Home
    ✓ should have an input
.    ✓ should have a list of computer scientists
.    ✗ should add a name to the list using the form
      - Expected 'Edsger Dijkstra
Donald Knuth
Alan Turing
Grace Hopper' to equal 'Edsger Dijkstra
Donald Knuth
Alan Turing
Grace Hopper
Tim Berners-Lee'.
F 
```

**Comment by mgechev** [comment]
@Shyam-Chen awesome! The build is passing now. Are we ready to merge?

File: package.json [hunk 1]
```diff
@@ -28,7 +28,7 @@
     "serve.prod": "gulp serve.prod --color",
     "start": "gulp serve.dev --color",
     "tasks.list": "gulp --tasks-simple --color",
-    "test": "gulp test --color",
+    "test": "gulp test --color && gulp build.prod --color && gulp build.js.e2e --color && gulp e2e --color",
```
Comment (mgechev) [inline package.json]: Lets move `gulp e2e` to a script called `e2e` and thus:

```bash
npm run e2e
```

Most developers will expect to run only karma when run `npm t`.

File: gulpfile.ts [hunk 2]
```diff
@@ -113,3 +113,33 @@ gulp.task('test', (done: any) =>
   runSequence('build.test',
               'karma.start',
               done));
+
+// try it
+import * as express from 'express';
```
Comment (mgechev) [inline gulpfile.ts]: Would you move this task definition in `/tools/tasks/seed/e2e.ts`? You'll get the exact same result.

File: package.json (hunk 1 above)
Comment (mgechev) [inline package.json]: Also, we can have `npm run tests.all` which runs both `npm test && npm run e2e`. This command can be used in CI>

**Comment by mgechev** [comment]
@Shyam-Chen really awesome job! Left some minor comments.

**Comment by Shyam-Chen** [comment]
OK

**Comment by mgechev** [comment]
LGMT! @Shyam-Chen ready for merge?

**Comment by Shyam-Chen** [comment]
merge

**Pull request merged**